server/src/main/java/org/elasticsearch/index/fielddata/SortingNumericDocValues.java

```diff
@@ -51,16 +58,40 @@ protected int compare(int i, int j) {
                 return Long.compare(values[i], values[j]);
             }
         };
+        this.circuitBreakerConsumer = circuitBreakerConsumer;
+        // account for initial values size of 1
+        this.circuitBreakerConsumer.accept(Long.BYTES);
     }
 
     /**
      * Set the {@link #docValueCount()} and ensure that the {@link #values} array can
      * store at least that many entries.
      */
     protected final void resize(int newSize) {
+        int oldValuesLength = values.length;
+        long oldValuesSizeInBytes = values.length * Long.BYTES;
+
+        // If array is expected to grow, increment the circuit breaker
+        // to include both the additional bytes used by the grown array
+        // as well as the overhead of keeping both arrays in memory while
+        // copying.
+        if (newSize > values.length) {
+            int newValuesLength = ArrayUtil.oversize(newSize, Long.BYTES);
+            long bytesDiff = (newValuesLength - oldValuesLength) * Long.BYTES;
+            circuitBreakerConsumer.accept(bytesDiff);
```

Comment:
Why not increment by the new size here and then decrement by the old size when we're done?

Reply:
I guess I was trying to keep consistent with the old logic, but I can see how this iterative addition can also be somewhat obtuse in this new logic.